File: c++/primeRange.cpp
#include<iostream>
using namespace std;

int main(){
    int n;
    cout << "Enter limit: ";
    cin >> n;
    bool prime[n+1];

    for(int i=0; i<=n; i++){
        prime[i] = true;
    } 

    for(int divisor = 2; divisor * divisor <= n; divisor++){
        if(prime[divisor]){
            for(int i = 2 * divisor; i<=n; i+=divisor){
                prime[i] = false;
            }
        }
    }

    for(int i=2; i<=n; i++){
        if(prime[i]){
            cout << " " << i;
        }
    }

    return 0;
}   

File: Python/primeRange.py
def printPrime():
    n = int(input("Enter Limit: \n"))
    prime = [True] * (n+1)

    divisor = 2
    while((divisor * divisor) <= n):
        
        if(prime[divisor]):
            i = 2 * divisor
            while(i<=n):
                prime[i] = False
                i += divisor
        
        divisor += 1

    j = 2

    while j <= n:
        if(prime[j]):
            print(j)
        j += 1


printPrime()

